frontend/src/scenes/insights/InsightTabs/RetentionTab.tsx

```diff
@@ -58,7 +58,7 @@ export function RetentionTab({ annotationsToCreate }: BaseTabProps): JSX.Element
                                 hideMathSelector
                                 hideFilter
                                 buttonCopy="Add graph series"
-                                filters={filters}
+                                filters={{ events: [filters.target_entity] }} // retention filters use target and returning entity instead of events
```

Comment:
Nit: Is it possible to use a selector in retentionLogic for this construction instead? It think it would keep our frontend a bit cleaner

Reply:
yessir!